### PR TITLE
[ingress-nginx] DO NOT MERGE. Move ingress-nginx to src artifact

### PR DESCRIPTION
**Pull request by name212**
## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: <kebab-case of a module name> | <1st level dir in the repo>
type: fix | feature | chore
summary: <ONE-LINE of what effectively changes for a user>
impact: <what to expect for users, possibly MULTI-LINE>, required if impact_level is high ↓
impact_level: default | high | low
```

### CODE REVIEW

File: modules/402-ingress-nginx/images/controller-1-10/werf.inc.yaml [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+      - '**/*'
+- add: /{{ $.ModulePath }}modules/{{ $.ModulePriority }}-{{ $.ModuleName }}/images/{{ $.ImageName }}/nginx-chroot-wrapper.sh
+  to: /src/nginx-chroot-wrapper.sh
+  stageDependencies:
+    install:
+      - '**/*'
+shell:
+  install:
+  {{- include "alpine packages proxy" . | nindent 2 }}
+  - apk --no-cache add patch
```
Comment (RomanenkoDenys) [inline modules/402-ingress-nginx/images/controller-1-10/werf.inc.yaml:30]: can we use git apply here ?

File: modules/402-ingress-nginx/images/controller-1-10/werf.inc.yaml [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+  - apt-get install -y gcc lua5.1-devel lua5.1 lua5.1-luarocks
+  - apt-get install -y liblua5.1-devel --download-only
+  setup:
+  - cd /src/lua-protobuf
+  - luarocks-5.1 make rockspecs/lua-protobuf-scm-1.rockspec
+  - cd /src/lua-iconv
+  - luarocks-5.1 install lua-iconv-7-3.src.rock
+---
+artifact: {{ .ModuleName }}/{{ $.ImageName }}-controller-artifact
+from: {{ .Images.BASE_GOLANG_22_ALPINE }}
```
Comment (RomanenkoDenys) [inline modules/402-ingress-nginx/images/controller-1-10/werf.inc.yaml:92]: use please BASE_GOLANG_23_ALPINE

File: modules/402-ingress-nginx/images/kruise-state-metrics/werf.inc.yaml [hunk 3]
```diff
@@ -1,23 +1,36 @@
...
     stageDependencies:
       install:
         - '**/*'
+shell:
+  install:
+    - mkdir -p /src
+    - git clone {{ $.SOURCE_REPO }}/openkruise/kruise-state-metrics.git /src
+    - cd /src
+    - git checkout {{ $commit }}
+    - git apply /patches/*.patch --verbose;
```
Comment (RomanenkoDenys) [inline modules/402-ingress-nginx/images/kruise-state-metrics/werf.inc.yaml:18]:
```suggestion
    - git apply /patches/*.patch --verbose
```

File: modules/402-ingress-nginx/images/kruise-state-metrics/werf.inc.yaml [hunk 4]
```diff
@@ -1,23 +1,36 @@
...
 git:
-  - add: /{{ $.ModulePath }}/modules/402-{{ $.ModuleName }}/images/{{ $.ImageName }}/patches
+  - add: /{{ $.ModulePath }}modules/{{ $.ModulePriority }}-{{ $.ModuleName }}/images/{{ $.ImageName }}/patches
     to: /patches
     stageDependencies:
       install:
         - '**/*'
+shell:
+  install:
+    - mkdir -p /src
```
Comment (RomanenkoDenys) [inline modules/402-ingress-nginx/images/kruise-state-metrics/werf.inc.yaml:14]: /src dir exists in the src-artfact, we shouldn't create it by hand

File: modules/402-ingress-nginx/images/kruise/werf.inc.yaml [hunk 5]
```diff
@@ -1,25 +1,34 @@
...
+    to: /patches
     stageDependencies:
-      setup:
+      install:
         - '**/*'
-    includePaths:
-      - patches
+shell:
+  install:
+    - mkdir -p /src
```
Comment (RomanenkoDenys) [inline modules/402-ingress-nginx/images/kruise/werf.inc.yaml:13]: same

File: modules/402-ingress-nginx/images/nginx-exporter/werf.inc.yaml [hunk 6]
```diff
@@ -1,15 +1,25 @@
...
 - fromPath: ~/go-pkg-cache
   to: /go/pkg
 shell:
   install:
-  - mkdir /src
   - cd /src
-  - git clone --depth 1 --branch v0.8.0 {{ $.SOURCE_REPO }}/nginxinc/nginx-prometheus-exporter.git .
-  - go mod edit -go 1.20
+  - go mod edit -go 1.23
   - go get github.com/prometheus/client_golang@v1.17.0
```
Comment (RomanenkoDenys) [inline modules/402-ingress-nginx/images/nginx-exporter/werf.inc.yaml:23]: we should move that to patch

File: modules/402-ingress-nginx/images/controller-1-10/werf.inc.yaml (hunk 1 above)
Comment (name212) [reply modules/402-ingress-nginx/images/controller-1-10/werf.inc.yaml:30]: no git apply fail with error

File: modules/402-ingress-nginx/images/kruise-state-metrics/werf.inc.yaml (hunk 3 above)
Comment (name212) [reply modules/402-ingress-nginx/images/kruise-state-metrics/werf.inc.yaml:18]: done

File: modules/402-ingress-nginx/images/kruise/werf.inc.yaml (hunk 5 above)
Comment (name212) [reply modules/402-ingress-nginx/images/kruise/werf.inc.yaml:13]: done

File: modules/402-ingress-nginx/images/controller-1-10/werf.inc.yaml (hunk 2 above)
Comment (name212) [reply modules/402-ingress-nginx/images/controller-1-10/werf.inc.yaml:92]: done

File: modules/402-ingress-nginx/images/nginx-exporter/werf.inc.yaml (hunk 6 above)
Comment (name212) [reply modules/402-ingress-nginx/images/nginx-exporter/werf.inc.yaml]: done

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

**Pull request merged**